znvis/particle/particle.py

```
@@ -124,8 +128,11 @@ def construct_mesh_list(self):
                 if self.director is not None:
                     self.director = self.director[np.newaxis, :, :]
 
-        except ValueError:
-            raise ValueError("There is no data for these particles.")
+        except IndexError:
+            raise IndexError("The provided data has an incompatible shape.")
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)